dmn-engine/src/main/scala/org/camunda/dmn/evaluation/ListEvaluator.scala

```diff
@@ -1,26 +1,27 @@
 package org.camunda.dmn.evaluation
 
 import scala.collection.JavaConverters._
-
 import org.camunda.dmn.DmnEngine._
 import org.camunda.dmn.FunctionalHelper._
-import org.camunda.bpm.model.dmn.instance.{List, Expression}
+import org.camunda.bpm.model.dmn.instance.{Expression, List}
 import org.camunda.dmn.parser.{ParsedDecisionLogic, ParsedList}
-import org.camunda.feel.syntaxtree.{Val, ValList}
+import org.camunda.feel.syntaxtree.{Val, ValError, ValList}
 import org.camunda.dmn.Audit.SingleEvaluationResult
 
 class ListEvaluator(
     eval: (ParsedDecisionLogic, EvalContext) => Either[Failure, Val]) {
 
   def eval(list: ParsedList, context: EvalContext): Either[Failure, Val] = {
 
-    mapEither(list.entries, (expr: ParsedDecisionLogic) => eval(expr, context)).right
-      .map(ValList)
-      .map { result =>
+    mapEither(list.entries, (expr: ParsedDecisionLogic) => eval(expr, context))
+      .map(ValList) match {
+      case r @ Right(result) =>
         context.audit(list, SingleEvaluationResult(result))
-
-        result
-      }
+        r
+      case l @ Left(failure) =>
+        context.audit(list, SingleEvaluationResult(ValError(failure.message)))
+        l
+    }
```

Comment:
With the suggested audit function, this can be simplified to:

```
    val result = mapEither(list.entries, (expr: ParsedDecisionLogic) => eval(expr, context))
      .map(ValList)

    context.audit(result, list)
    result
```